code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -233,3 +233,147 @@
+	keybind_signal = COMSIG_XENOABILITY_RAGE
+	///Determines the power of Rage's many effects. Power scales inversely with the Ravager's HP; min 0.25 at 50% of Max HP, max 1 while in negative HP. 0.5 and above triggers especial effects.
+	var/rage_power
+	///Determines the Sunder to impose when Rage ends
+	var/rage_sunder
+	///Determines the Plasma to remove when Rage ends
+	var/rage_plasma
+
+/datum/action/xeno_action/rage/on_cooldown_finish()
+	to_chat(owner, "<span class='xenodanger'>We are able to enter our rage once again.</span>")
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+/datum/action/xeno_action/rage/can_use_action(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/ravager/rager = owner
+
+	if(rager.health > rager.maxHealth * RAVAGER_RAGE_MIN_HEALTH_THRESHOLD) //Need to be at 50% of max hp or lower to rage
+		if(!silent)
+			to_chat(rager, "<span class='xenodanger'>Our health isn't low enough to rage! We must take [rager.health - (rager.maxHealth * RAVAGER_RAGE_MIN_HEALTH_THRESHOLD)] more damage!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/rage/action_activate()
+	var/mob/living/carbon/xenomorph/ravager/X = owner
+
+	rage_power = (1-(X.health/X.maxHealth)) * RAVAGER_RAGE_POWER_MULTIPLIER //Calculate the power of our rage; scales with difference between current and max HP
+
+	if(X.health < 0) //Gain additional rage with negative HP; gain + 0.0075 rage power per point of negative HP
+		rage_power += X.health * RAVAGER_RAGE_NEGATIVE_HP_POWER_MULTIPLIER
+
+	rage_power = min(1, rage_power) //Cap rage power so that we don't get way too insane.
+
+	var/rage_power_radius = CEILING(rage_power * 7, 1) //Define radius of the SFX
+
+	X.visible_message("<span class='danger'>\The [X] becomes frenzied, bellowing with a shuddering roar!</span>", \
+	"<span class='highdanger'>We bellow as our fury overtakes us! RIP AND TEAR!</span>")
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > RAVAGER_RAGE_SUPER_RAGE_THRESHOLD) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+	for(var/turf/affected_tiles AS in RANGE_TURFS(rage_power_radius, X.loc))
+		affected_tiles.Shake(4, 4, 1 SECONDS) //SFX
+
+	for(var/mob/living/L AS in cheap_get_living_near(X, rage_power_radius)) //Roar that applies cool SFX
+		if(L.stat == DEAD || !L.hud_used) //We don't care about the dead
+			continue
+
+		shake_camera(L, 1 SECONDS, 1)
```

Comment:
camera shake is way too annoying and vomit prone, so better not have this, unless you find something else to replace this with

Reply:
eh, it's 1 second and triggers only when the ravager goes full super sayjack at negative HP via Endure